### PR TITLE
Fix data race in log message of cached buffer

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp [hunk 1]
```diff
@@ -1058,34 +1078,13 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
         available(),
         current_read_range.toString(),
         file_offset_of_buffer_end,
-        implementation_buffer->getFileOffsetOfBufferEnd(),
```
Comment (kssenii) [inline src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp:1061]: a different fix could be to move `file_segment.completePartAndResetDownloader` after this log message, but I want this logging to be after - to see the completed state in log, so removed the cause of the data race this way

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit ea9d0f6c3c94ae6fef1bb552958455c7ab9b25e7 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50723/ea9d0f6c3c94ae6fef1bb552958455c7ab9b25e7/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by SmitaRKulkarni — APPROVED** [review]
LGTM

**Comment by kssenii** [comment]
Failed tests not related to changes:

> Stateless tests (debug) [2/5] — fail: 1

02703_max_local_write_bandwidth

> Stateless tests (debug, s3 storage) [1/6] — fail: 1

02479_mysql_connect_to_self

**Pull request merged**